src/components/molecules/OakModalCenter/OakModalCenter.tsx

```diff
@@ -0,0 +1,257 @@
+import React, {
+  HTMLAttributes,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { createPortal } from "react-dom";
+import { FocusOn } from "react-focus-on";
+import { Transition, TransitionStatus } from "react-transition-group";
+import styled from "styled-components";
+
+import { InternalShadowRoundButton } from "../InternalShadowRoundButton";
+
+import { OakBox, OakFlex, OakFlexProps, oakBoxCss } from "@/components/atoms";
+import { parseSpacing } from "@/styles/helpers/parseSpacing";
+
+export type OakModalCenterProps = {
+  /**
+   * The content of the modal. Use with `<OakModalCenterBody>` for best results.
+   */
+  children: ReactNode;
+  /**
+   * Determines whether to show the modal or not
+   */
+  isOpen: boolean;
+  /**
+   * Override HTMLAttributes & OakFlex props for the modal container
+   */
+  modalFlexProps?: Partial<OakFlexProps & HTMLAttributes<Element>>;
+  /**
+   * Override HTMLAttributes & OakFlex props for the backdrop container
+   */
+  backdropFlexProps?: Partial<OakFlexProps & HTMLAttributes<Element>>;
+  /**
+   * If true, clicking the backdrop will not call onClose
+   */
+  disableBackdropClick?: boolean;
+  /**
+   * If true, pressing the escape key will not call onClose
+   */
+  disableEscapeKey?: boolean;
+  /**
+   * If true, the close button will be hidden
+   */
+  hideCloseButton?: boolean;
+  /**
+   * Called when the modal is closed via the close button, backdrop click, or escape key
+   */
+  onClose: () => void;
+  /**
+   * The DOM container to render the modal portal into.
+   */
+  domContainer?: Element;
+  /**
+   * Slot for the footer of the modal
+   */
+  footerSlot?: ReactNode;
+};
+
+const FocusOnBox = styled(FocusOn)`
+  ${oakBoxCss}
+`;
+
+const FadeInFlex = styled(OakFlex)<{ $state: TransitionStatus }>`
+  opacity: ${({ $state }) => {
+    switch ($state) {
+      case "entered":
+      case "entering":
+        return "1";
+      default:
+        return "0";
+    }
+  }};
+  transition: opacity 0s;
```

Comment:
Transition is 0. Originally I had a nice fade, but that doesn't work with the backdrop blur effect. It only turns blurred when the animation finishes